app/models/transaction.rb

```diff
@@ -0,0 +1,21 @@
+class Transaction < ActiveRecord::Base
+  before_save :increment_version, :create_version, if: :changed?
+
+  validates :uid, :booked_at, :booking_at, :booking_status, :delivery_partner,
+            presence: true
+  validates :uid, uniqueness: true
+  validates :delivery_partner, inclusion: { in: Partners.delivery_partners }
+
+  scope :bookings, ->(end_point) { where(['booked_at <= ?', end_point]).where(cancelled: false) }
```

Comment:
Pretty sure you don't need to wrap the parameters in an array here, and in line 11.